### PR TITLE
fix: use employer or same site when doing employer actions

### PR DESCRIPTION
**Pull request by fw-noel**
This is a fix for https://freshworks.atlassian.net/browse/HCAP-1308 but serves in two ways:

When posting an employer action, there's a touch less reliance on Employer Id of a status. 1308 was caused when the `employer_id` was different from the user trying to archive. In `participant-status.js`, we see the cause:
```
    const existingCurrentStatus = currentStatusId
      ? await tx[collections.PARTICIPANTS_STATUS].findOne({ id: currentStatusId })
      : await tx[collections.PARTICIPANTS_STATUS].findOne({
          participant_id: participantId,
          current: true,
          employer_id: employerId,
        });
```

~~That is when no currentStatusId is passed in. By getting the status id first, we not pass it in here properly, but we also check that the accessing user has the right permissions for the participant.~~

We're keeping it at no status passed in, but there's still less reliance on the hiring user being the same person.

### CODE REVIEW

File: server/routes/participant.js [hunk 1]
```diff
@@ -430,8 +431,10 @@ employerActionsRouter.post(
     if (!participant) {
       return res.status(400).json({ message: 'Could not find participant' });
     }
+    const currentStatus = await getParticipantCurrentStatus(participantId);
+    const currentSite = site || currentStatus?.data?.site;
```
Comment (arranfw) [inline server/routes/participant.js:435]: Should this be `currentSiteId`?

File: server/services/participants.js [hunk 2]
```diff
@@ -127,6 +127,14 @@ const getParticipantByID = async (participantInfo) => {
   return participant;
 };
 
+const getParticipantCurrentStatus = async (participantId) => {
+  const participant = await dbClient.db[collections.PARTICIPANTS_STATUS].findOne({
```
Comment (arranfw) [inline server/services/participants.js:131]: Does this take into account that participants can have more than one current status as a time? Does it matter?

**Review by arranfw — DISMISSED** [review]

File: server/tests/participant-status.test.js [hunk 3]
```diff
@@ -168,23 +168,23 @@ describe('Test Participant status data model and service', () => {
       city: 'Test City 1030',
     });
 
-    const emp1 = v4();
-    const emp2 = v4();
-    const emp3 = v4();
+    const emp1 = { id: v4(), sites: [site1.siteId, site2.siteId] };
```
Comment (fw-noel) [inline server/tests/participant-status.test.js:171]: I think these tests make more sense. In the original, emp1 was prospecting the same participant for two sites, which I don't believe is a scenario.

File: client/src/utils/prettifyStatus.js [hunk 4]
```diff
@@ -25,7 +25,7 @@ export const getParticipantStatusData = (status, tabValue, isMoH, participantInf
   const isPendingAcknowledgement = status.includes('pending_acknowledgement');
   const isWithdrawn = status.includes('withdrawn');
   const isArchived = status.includes('archived');
-  const rosStartDate = isROS ? participantInfo.rosStatuses[0].data.date : null;
+  const rosStartDate = isROS ? participantInfo?.rosStatuses[0]?.data?.date : null;
```
Comment (fw-noel) [inline client/src/utils/prettifyStatus.js:28]: This is unrelated to HCAP-1308, I just thought it was a minor change that I could piggyback here. Just some null checks. I couldn't reproduce Akhil's crashing issue, but this seems like a good idea regardless.

**Review by arranfw — APPROVED** [review]

**Review by alva-fresh — APPROVED** [review]
Looks good!

**Review by taraepp — APPROVED** [review]
